index.d.ts

```diff
@@ -0,0 +1,97 @@
+import * as Cookies from 'cookies'
+import type Firebase from 'firebase'
+import type {
+  GetServerSidePropsContext,
+  GetServerSidePropsResult,
+  NextApiRequest,
+  NextApiResponse,
+} from 'next'
+import type { ComponentType } from 'react'
+import type { ParsedUrlQuery } from 'querystring'
+
+type GetSSRProps<P extends { [key: string]: any } = { [key: string]: any },
+  Q extends ParsedUrlQuery = ParsedUrlQuery> = (
+  context: GetServerSidePropsContext<Q> & { AuthUser: AuthUser },
+) => Promise<GetServerSidePropsResult<P>>
+
+export enum AuthAction {
+  RENDER = 'render',
+  SHOW_LOADER = 'showLoader',
+  RETURN_NULL = 'returnNull',
+  REDIRECT_TO_LOGIN = 'redirectToLogin',
+  REDIRECT_TO_APP = 'redirectToApp'
+}
+
+export interface AuthUser {
+  id: string | null;
+  email: string | null;
+  emailVerified: boolean;
+  getIdToken: () => Promise<string | null>;
+  clientInitialized: boolean;
+  firebaseUser: Firebase.User | null;
+  signOut: () => Promise<void>;
+}
+
+interface AuthUserContext extends AuthUser {
+  serialize: (opts?: { includeToken?: boolean }) => string;
+}
+
+interface InitConfig {
+  authPageURL?: string;
+  appPageURL?: string;
+  loginAPIEndpoint?: string;
+  logoutAPIEndpoint?: string;
+  tokenChangedHandler?: (user: AuthUser) => void;
+  firebaseAdminInitConfig: {
+    credential: {
+      projectId: string;
+      clientEmail: string;
+      privateKey: string;
+    };
+    databaseURL: string;
+  };
+  firebaseClientInitConfig: {
+    apiKey: string;
+    authDomain: string;
+    databaseURL: string;
+    projectId: string;
+  };
+  cookies: Cookies.Option & Cookies.SetOption;

```

Comment:
I'm inclined to explicitly add the required `name` field to call out that its behavior differs from the cookies library (it's a base name for multiple cookies). Open to skipping this if you don't think it's useful.

Reply:
I started using this library less than two days ago, so I will defer to you on that. I'll add it. :-)